lua/acf/shared/armor/rubber.lua

```diff
@@ -37,36 +44,45 @@ if SERVER then
 		local effectiveness = Material.effectiveness
 		local resiliance	= Material.resiliance
 
+		local ductilityvalue = (Entity.ACF.Ductility or 0) * 1.25 --The ductility value of the armor. Outputs 1 to -1 depending on max ductility
+		local ductilitymult    = 2 / (2 - ductilityvalue * 1.5) -- Direct damage multiplier based on ductility.
+
 		armor	= armor ^ curve
 		losArmor	= losArmor ^ curve
 
+
 		--=========================================================================================================\
 		--------------------------------------------------------- For HEAT shells & Spall -------------------------->
 		--=========================================================================================================/
-		if Type == "HEAT" or Type == "THEAT" or Type == "HEATFS" or Type == "THEATFS" or Type == "Spall" then
+		if Type == "HEAT" or Type == "THEAT" or Type == "HEATFS" or Type == "THEATFS" or Type == "Spall" then --Gotta love shelltype bloat. P L E A S E  Stop appeasing the bloaters.
```

Comment:
```lua
local validTypes = {
    ["HEAT"] = true,
    ["THEAT"] = true,
    ["HEATFS"] = true,
    ["THEATFS"] = true,
    ["Spall"] = true
}

local Type = "HEAT" -- Replace with your actual Type value

if validTypes[Type] then
    -- Type is valid
    -- Your code here
else
    -- Type is not valid
    -- Handle the case where Type is not in the list
end
```

Reply:
You could solve cases like this by just adding another field to ammo types, ie `AMMO.DamageMultiplier`, so in cases where you need that value you just look for it or default it to something else in case it doesn't exist (`Multiplier = AMMO.DamageMultiplier or 1`) rather than having to maintain all these lookup tables scattered around.